oap-server/server-receiver-plugin/envoy-metrics-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/envoy/MetricServiceGRPCHandler.java

```diff
@@ -79,17 +93,15 @@ public void onNext(StreamMetricsMessage message) {
                     isFirst = false;
                     StreamMetricsMessage.Identifier identifier = message.getIdentifier();
                     Node node = identifier.getNode();
-                    if (node != null) {
-                        String nodeId = node.getId();
-                        if (!StringUtil.isEmpty(nodeId)) {
-                            serviceInstanceName = nodeId;
-                        }
-                        String cluster = node.getCluster();
-                        if (!StringUtil.isEmpty(cluster)) {
-                            serviceName = cluster;
-                            if (serviceInstanceName == null) {
-                                serviceInstanceName = serviceName;
-                            }
+                    String nodeId = node.getId();
+                    if (!StringUtil.isEmpty(nodeId)) {
+                        serviceInstanceName = nodeId;
+                    }
+                    String cluster = node.getCluster();
+                    if (!StringUtil.isEmpty(cluster)) {
+                        serviceName = cluster;
+                        if (serviceInstanceName == null) {
+                            serviceInstanceName = serviceName;
```

Comment:
This is the logic we used to do, but I think we should share the ALS logic. Then in future integration, the entity names of sidecar metrics will be as same as the traffic related metrics.

Reply:
The service names in metrics service have a service group prefix (`istio-dp::`) while those in ALS don't have, do we want to add service group in ALS as well?